python/ray/experimental/signal.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import hashlib
+import time
+
+import ray
+
+SYSTEM_ID_STRING = ray.ray_constants.ID_SIZE * b"\xaa"
+SYSTEM_ID = ray.ObjectID(SYSTEM_ID_STRING)
+
+SIG_USER = 100
+
+class Signal(object):
+    """Signal object"""
+    def __init__(self, type, value):
+        self.sig_type = type
+        self.sig_value = value
+    def value(self):
+        return self.sig_value
+    def type(self):
+        return self.sig_type
+
+def _get_signal_object_id_from_binary(source_id, counter):
+    m = hashlib.sha1()
+    m.update(source_id)
+    m.update(bytes(counter))
+    return ray.ObjectID(m.digest())
+
+def _get_signal_object_id(source_id, counter):
+    if  type(source_id) is ray.actor.ActorHandle:
+        return _get_signal_object_id_from_binary(source_id._actor_id.id(), counter)
```

Comment:
Use the actor creation task ID instead of the actor ID. This will be `ComputeTaskId(source_id._ray_actor_creation_dummy_object_id)`. `ComputeTaskId` is implemented in C++ but also needs a Python extension.

Reply:
Actually, an extension for `ComputeTaskId` already exists: [code](https://github.com/ray-project/ray/blob/master/src/ray/raylet/lib/python/common_extension.cc#L825). This is a good example for how you should implement the Python extension for `ComputeReturnId`.